src/main/java/uk/gov/ch/oauth/Oauth2.java

```diff
@@ -78,11 +81,12 @@ private Payload oauth2DecodeState(final String state) {
     }
 
     @Override
-    public boolean isValid(final String state, final String code) {
+    public boolean isValid(final String state, final String code,
+            final HttpServletResponse httpServletResponse) {
```

Comment:
If this is a function that performs logic and manipulates data calling it 'isValid' is probably misleading as it sounds like a simple getter function. Perhaps this should be called 'validate' or something similar.

Reply:
I don't know why this appeared twice?